### PR TITLE
Initial part of Auto-Scaling on Google Cloud

### PR DESCRIPTION
**Pull request by rimijoker**
## Description

I tried to implement a part of Auto-Scaling on GCP, now one can spin-up a compute instance just by using a few lines of code. I have added a README.md for further instructions.

To use this you will have to use a python script as I used the subprocess module which doesn't work in Jupyter Notebooks. If somebody knows how can we make it work in Jupyter Notebooks please let me know :)

Fixes(Initial Implementation, more code on the way) #3099 (issue)

## Type of change

Please mark the options that are relevant.

- [ ] Added/Modified tutorials
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

* [x] I did follow the [contribution guidelines](https://github.com/OpenMined/PySyft/blob/master/CONTRIBUTING.md)
* [x] I have added tests for my changes
* [x] I have commented my code following [Google style](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html).

### CODE REVIEW

File: syft/federated/autoscale/gcloud.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        with open("main.tf.json", "w") as main_config:
+            json.dump(self.config, main_config, indent=2, sort_keys=False)
+        # subprocess.call("terraform plan", shell=True)
+        subprocess.call("terraform apply", shell=True)
+
+    def destroy(self):
+        """
+        args:
+        """
+        os.remove(self.credentials)
```
Comment (hericlesme) [inline syft/federated/autoscale/gcloud.py:58]: Terraform needs credentials to destroy. Remove the link to the file after calling destroy.

Comment (rimijoker) [reply syft/federated/autoscale/gcloud.py]: Fixed it by deleting credentials after terraform destroy.

**Comment by hericlesme** [comment]
@rimijoker could you move these changes to the `grid` package?

**Comment by rimijoker** [comment]
> @rimijoker could you move these changes to the `grid` package?

Sure

**Comment by rimijoker** [comment]
> @rimijoker could you move these changes to the `grid` package?

@hericlesme It's done.

**Pull request merged**